src/main/java/hu/oe/nik/szfmv/automatedcar/model/CollisionDetector.java

```diff
@@ -0,0 +1,13 @@
+package hu.oe.nik.szfmv.automatedcar.model;
+
+import java.awt.*;
+import java.awt.geom.Area;
+import java.util.function.BiFunction;
+
+public class CollisionDetector implements BiFunction<Polygon, Polygon, Boolean> {
```

Comment:
Instead of `BiFunction<Polygon, Polygon, Boolean>` you can use `BiPredicate<Polygon, Polygon>`, which is a bit more descriptive and simple. Implementation of the interface can remain the same (possibly with another method name, but same logic).